components/server/src/projects/projects-service.spec.db.ts

```diff
@@ -0,0 +1,208 @@
+import { Organization, User } from "@gitpod/gitpod-protocol";
+import { ResponseError } from "vscode-ws-jsonrpc";
+
+const expect = chai.expect;
+
+describe("ProjectsService", async () => {
+    let container: Container;
+    let owner: User;
+    let stranger: User;
+    let org: Organization;
+
+    beforeEach(async () => {
+        container = testContainer.createChild();
+        container.load(
+            new ContainerModule((bind) => {
+                bind(OrganizationService).toSelf().inSingletonScope();
+                bind(ProjectsService).toSelf().inSingletonScope();
+                bind(Config).toConstantValue({});
+                bind(AuthProviderService).toSelf().inSingletonScope();
+                bind(IAnalyticsWriter).toConstantValue(NullAnalyticsWriter);
+                // hostcontext
+                bind(HostContainerMapping).toSelf().inSingletonScope();
+                bind(HostContextProviderFactory)
+                    .toDynamicValue(({ container }) => ({
+                        createHostContext: (config: AuthProviderParams) =>
+                            HostContextProviderImpl.createHostContext(container, config),
+                    }))
+                    .inSingletonScope();
+                bind(HostContextProvider).to(HostContextProviderImpl).inSingletonScope();
+
+                // auth
+                bind(SpiceDBClient)
+                    .toDynamicValue(() => {
+                        const token = uuidv4();
+                        return v1.NewClient(token, "localhost:50051", v1.ClientSecurity.INSECURE_PLAINTEXT_CREDENTIALS)
+                            .promises;
+                    })
+                    .inSingletonScope();
+                bind(SpiceDBAuthorizer).toSelf().inSingletonScope();
+                bind(Authorizer).toSelf().inSingletonScope();
+            }),
+        );
+        Experiments.configureTestingClient({
+            centralizedPermissions: true,
+        });
+        const userDB = container.get<UserDB>(UserDB);
+        owner = await userDB.newUser();
+        stranger = await userDB.newUser();
+        const orgService = container.get(OrganizationService);
+        org = await orgService.createOrganization(owner.id, "my-org");
+    });
+
+    afterEach(async () => {
+        // Clean-up database
+        const typeorm = testContainer.get(TypeORM);
+        const dbConn = await typeorm.getConnection();
+        await dbConn.getRepository(DBTeam).delete({});
+        await dbConn.getRepository(DBProject).delete({});
+        const repo = (await typeorm.getConnection()).getRepository(DBUser);
+        await repo.delete(owner.id);
+        await repo.delete(stranger.id);
+    });
+
+    it("should let org owners create projects", async () => {
+        const ps = container.get(ProjectsService);
+        const project = await createTestProject(ps, org, owner);
+        expect(project).to.not.be.undefined;
+    });
```

Comment:
I think here, we can combine the unit test for a perm check, and the test for the `createProject` and check that we get the right values back.

I think in general, for the happy cases we don't have to include the "is allowed" in the test description, but for the negative cases it would make sense